applications/CoSimulationApplication/python_scripts/coupling_operations/compute_boundary_force.py

```diff
@@ -0,0 +1,163 @@
+                    result_msg += ', '.join([a + b for a, b in zip(res_labels, output_values)])
+                    cs_tools.cs_print_info(self._ClassName(), result_msg)
+                
+                if(self.write_output_file):
+                    self.output_file.write(' '.join(output_values) + '\n')
+    
+    def _EvaluateGlobalForces(self):
+        # vel_x, vel_y, vel_z
+        velocity = [0.0, 0.0, 0.0]
+        sum_forces = [0.0, 0.0, 0.0]
+        pressure_list = [0.0]
+
+        # aktuellen Zeitstep zurück setzen - warum ist das nötig ?
+        # TODO !!
+        for node in self.model_part.Nodes:
+            node.SetSolutionStepValue(KM.REACTION_X, 0, 0)
+            node.SetSolutionStepValue(KM.REACTION_Y, 0, 0)
+            node.SetSolutionStepValue(KM.REACTION_Z, 0, 0)
+
+        for element in self.model_part.Elements:
+            geometry = element.GetGeometry()
+            nodes = element.GetNodes()
+            shape_functions_values = geometry.ShapeFunctionsValues()
+            area = geometry.Area()
+            unit_normal = geometry.UnitNormal()
+            # print normal info
+            debug_normal_info_mode = False
+            if debug_normal_info_mode:
+                self._PrintUnitNormalInfo(nodes, element, unit_normal)
+            
+            pressure = 0
+            c = 0
+            for node in nodes:
+                pressure_node = node.GetSolutionStepValue(KM.PRESSURE, 0)
+                force_node = unit_normal * (-1) * pressure_node * area * shape_functions_values[0, c] * self.width
+                pressure += pressure_node * shape_functions_values[0, c]
+
+                node.SetSolutionStepValue(KM.REACTION_X, 0, node.GetSolutionStepValue(KM.REACTION_X, 0) + force_node[0])
+                node.SetSolutionStepValue(KM.REACTION_Y, 0, node.GetSolutionStepValue(KM.REACTION_Y, 0) + force_node[1])
+                node.SetSolutionStepValue(KM.REACTION_Z, 0, node.GetSolutionStepValue(KM.REACTION_Z, 0) + force_node[2])
+                c += 1
+
+            force = unit_normal * pressure * area * self.width
+
+            for i in range(3):
+                sum_forces[i] += force[i]
+
+            pressure_list[0] += pressure
+        
+        if self.echo_level > 1:
+            info_msg = "Computed boundary forces for model part \"" + self.model_part_name  + "\" in solver: \"" + self.settings["solver"].GetString() + "\""
+            cs_tools.cs_print_info(self._ClassName(), info_msg)
+        
+        return velocity + sum_forces + pressure_list
+
+    def _PrintUnitNormalInfo(self, nodes, element, unit_normal):
+        # print normal debug information
+        import pdb
+        pdb.set_trace()
+        cs_tools.cs_print_info('Element info', element)
+        cs_tools.cs_print_info('Unit normal', unit_normal)
+        node_counter = 0
+        for node in nodes:
+            cs_tools.cs_print_info(str(node_counter) + ' Node Id', str(node.Id))
+            node_counter += 1
```

Comment:
this should not go to master as it is only for debuggign

Reply:
I eliminated this now.